src/core/lib/security/credentials/alts/check_gcp_environment_windows.cc

```diff
@@ -31,80 +31,70 @@
 #include <grpc/support/log.h>
 #include <grpc/support/sync.h>
 
-#define GRPC_ALTS_EXPECT_NAME_GOOGLE "Google"
-#define GRPC_ALTS_WINDOWS_CHECK_COMMAND "powershell.exe"
-#define GRPC_ALTS_WINDOWS_CHECK_COMMAND_ARGS \
-  "(Get-WmiObject -Class Win32_BIOS).Manufacturer"
-#define GRPC_ALTS_WINDOWS_CHECK_BIOS_FILE "windows_bios.data"
-
-const size_t kBiosDataBufferSize = 256;
-
-static bool g_compute_engine_detection_done = false;
-static bool g_is_on_compute_engine = false;
-static gpr_mu g_mu;
-static gpr_once g_once = GPR_ONCE_INIT;
-
 namespace grpc_core {
 namespace internal {
 
-bool check_bios_data(const char* bios_data_file) {
-  char* bios_data = read_bios_file(bios_data_file);
-  bool result = !strcmp(bios_data, GRPC_ALTS_EXPECT_NAME_GOOGLE);
-  remove(GRPC_ALTS_WINDOWS_CHECK_BIOS_FILE);
-  gpr_free(bios_data);
-  return result;
+bool check_bios_data(const char*) {
```

Comment:
Do we still need this function?

Reply:
Maybe not, I can remove it.  But it is declared here:

https://github.com/grpc/grpc/blob/2b06d76cc06e2fa4016713c1cabec510a4f89e87/src/core/lib/security/credentials/alts/check_gcp_environment.h#L36-L45

Which would leave a dangling declaration, and that cannot be removed because it is used in the Linux test:

https://github.com/grpc/grpc/blob/2b06d76cc06e2fa4016713c1cabec510a4f89e87/test/core/security/check_gcp_environment_linux_test.cc#L41

Thoughts?  I think it might be better to have a future PR remove the declaration and *only* declare it on the Linux test and not the header file.

Reply:
You are right, we cannot remove it yet. 
Let me or my teammates have a separate PR to move check_bios_data() to Linux only implementation.